Architecture.md

```diff
@@ -71,6 +71,116 @@
+
+In that sense, to prepare for future versions, the git controller is a **separated component** from the rest of the application, designed to be **stateless**, and only stores the repositories on the host machine. It does not store any information about the repositories, and only uses the **Project's id** to name the repository's directory. This way, the git controller can be scaled horizontally, and, if needed, the repositories can be stored on a dedicated storage server after adding a layer of authorization.
+
+Since it is separated it needs to communicate with the rest of the application. It was achieved using [gRPC](https://grpc.io/), which is a high performance RPC framework. It is used to create a bidirectional stream between the git controller and the rest of the application. The git controller can then receive commands from the rest of the application, and send back the result of the command.
+
+We made the choice to use gRPC instead of a REST API. Indeed, since it uses [protobuf](https://developers.google.com/protocol-buffers), it is easier to maintain and to evolve as the application grows. It is also more performant than a REST API, and is easier to use, since it is strongly typed. It's supported on different languages also makes it easier to use. It also limits communication problems between teams and services, since the communication is well defined and documented through the protobuf file.
+
+It was made in Rust for its high performance, reliability, robustness and memory safety using the [tonic](https://github.com/hyperium/tonic) framework, which is a Rust implementation of gRPC. It was chosen because it is the most used gRPC framework in Rust, and is well documented.
+
+Here's an overview of the interactions of the git controller:
+
+```mermaid
+sequenceDiagram
+    participant API
+    participant GitRepoManager
+    participant HostMachine
+
+    API ->> GitRepoManager: CreateProject / DeleteProject
+    GitRepoManager ->> HostMachine: git init / rm -rf project_id
+    HostMachine -->> GitRepoManager: Ok
+    GitRepoManager -->> API: Ok
+```
+
+##### Authentication
+
+Once the repositories are stored, the clients need to be able to push to them.
+
+In order to let the users push to thoses repositories, we need to give them access to the host machine. An easy solution would be to give them a ssh access to the host machine with servers like [sshd](https://www.openssh.com/). However, this would give them access to the whole host machine, which is not what we want. Indeed, we want to limit the access of the clients to their own repositories, and not to the whole host machine.
+
+One solution would be to use a git server like [gitea](https://gitea.io/en-us/) or [gitlab](https://about.gitlab.com/), which provide a full-fledged git service, with authentication and authorization. However, this would require to maintain a full git server, with possibly duplicated data, which is not the goal of PaaSTech.
```

Comment:
Please use `Gitea` and `Gitlab` with capital letters when naming them

Reply:
fixed: 153bb917b1d323bf3333819a461dc61343b0972c